delira/data_loading/data_manager.py

```diff
@@ -483,55 +203,17 @@ def update_state_from_dict(self, new_state: dict):
         self.data_loader_cls = new_state.pop("data_loader_cls",
                                              self.data_loader_cls)
         # update
-        new_sampler = new_state.pop("sampler", None)
+        new_sampler = new_state.pop("sampler_old", None)
         if new_sampler is not None:
             self.sampler = new_sampler.from_dataset(
-                self.dataset,
+                self.data,
                 **new_state.pop("sampling_kwargs", {}))
```

Comment:
I think, by popping sampler kwargs first, the creation of the sampler could be handled by the property setter

Reply:
This is obsolete by dynamically creating the sampler